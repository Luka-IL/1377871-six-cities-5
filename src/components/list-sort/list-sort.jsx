import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../store/action";

const ListSort = (props) => {
  const {changeSort} = props;

  const getSortValue = (evt) => {
    evt.preventDefault();
    changeSort(evt.target.value);
  };

  return (
    <select className="places__option places__options--custom places__sorting-type" id="places-sorting" onChange={getSortValue}>
      <option className="places__option" value="popular">Popular</option>
      <option className="places__option" value="to-high">Price: low to high</option>
      <option className="places__option" value="to-low">Price: high to low</option>
      <option className="places__option" value="top-rated">Top rated first</option>
    </select>
  );
};

ListSort.propTypes = {
  changeSort: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  changeSort(sort) {
    dispatch(ActionCreator.changeSort(sort));
  }
});

export {ListSort};
export default connect(null, mapDispatchToProps)(ListSort);
