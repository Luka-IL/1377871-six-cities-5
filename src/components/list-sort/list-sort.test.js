import React from "react";
import renderer from "react-test-renderer";
import {ListSort} from "./list-sort";

it(`Should ListSort render correctly`, () => {
  const tree = renderer
    .create(<ListSort
      changeSort={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
