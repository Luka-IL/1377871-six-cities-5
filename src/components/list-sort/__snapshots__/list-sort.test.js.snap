// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ListSort render correctly 1`] = `
<select
  className="places__option places__options--custom places__sorting-type"
  id="places-sorting"
  onChange={[Function]}
>
  <option
    className="places__option"
    value="popular"
  >
    Popular
  </option>
  <option
    className="places__option"
    value="to-high"
  >
    Price: low to high
  </option>
  <option
    className="places__option"
    value="to-low"
  >
    Price: high to low
  </option>
  <option
    className="places__option"
    value="top-rated"
  >
    Top rated first
  </option>
</select>
`;
